OADP-3327: DPA does not reconcile when credentials secret updated (#73975)
openshift/openshift-docs: modules/oadp-release-notes-1-2-3.adoc

[id="known-issues-1-2-3_{context}"]
== Known issues

> The following known issues have been included in the release of {oadp-short} 1.2.3:

Author: ```suggestion
The {oadp-short} 1.2.3 has the following known issue:
```

> The following known issues have been highlighted in the release of {oadp-short} 1.2.3:

Author: ```suggestion
The following known issues have been included in the release of {oadp-short} 1.2.3:
```

> The following issues have been highlighted as known issues in the release of OADP 1.2.3:

Author: ```suggestion
The following known issues have been highlighted in the release of {oadp-short} 1.2.3:
```

> .DPA does not reconcile when the credentials secret is updated

Reviewer: Please expand the _DPA_ acronym on the first usage.

See the [IBM Style (ISG)](https://www.ibm.com/docs/en/ibm-style?topic=grammar-abbreviations#spelled-out-forms) rule to spell out abbreviations for the first use.

> .DPA does not reconcile when the credentials secret is updated

Author: ```suggestion
. Data Protection Application (DPA) does not reconcile when the credentials secret is updated
```

> When we patch the cloud-credentials secret with empty data for the key "cloud", DPA does not reconcile and reconciles successfully.

Author: ```suggestion
There are no {oadp-short} specific labels or owner references on the `cloud-credentials` secret. Therefore, the Operator does not reconcile when you update the `cloud-credentials` secret. This means you could create a `cloud-credentials` secret with bad credentials, empty data for example, then the Operator will reconcile and create Backup Storage Location (BSL) and registry deployment with bad credentials. However, when you update  `cloud-credentials` secret with _good_  credentials, the Operator does not immediately reconcile to catch the new credentials. This issue is resolved in {oadp-short} 1.3.
```

> There are no {oadp-short} specific labels or owner references on the `cloud-credentials` secret. Therefore, the Operator does not reconcile when you update the `cloud-credentials` secret. This means you could create a `cloud-credentials` secret with bad credentials, empty data for example, then the Operator will reconcile and create Backup Storage Location (BSL) and registry deployment with bad credentials. However, when you update  `cloud-credentials` secret with _good_  credentials, the Operator does not immediately reconcile to catch the new credentials. This issue is resolved in {oadp-short} 1.3.

Reviewer: ```suggestion
Currently, the Operator does not reconcile when you update the `cloud-credentials` secret. This occurs because there are no {oadp-short} specific labels or owner references on the `cloud-credentials` secret. If you create a `cloud-credentials` secret with incorrect credentials, such as empty data, the Operator reconciles and creates a Backup Storage Location (BSL) and registry deployment with the empty data. As a result, when you update  `cloud-credentials` secret with the correct credentials, the Operator does not immediately reconcile to catch the new credentials. This issue is fixed in {oadp-short} 1.3.
```

Reviewer: I had a hard time connecting the customer experience to the cause of the known issue. I find that rearranging the order of the sentences makes it to easier to follow. Does this change the intended meaning too much?

Also, if there is no workaround for this issue in OADP 1.2.3, I think it is worth explicitly stating that the workaround is to update to 1.3. It is implied in what you have here, but I think this would benefit from being explicit.

The change from bad/good to incorrect/correct is not a blocker, it just sounds a little better to my ear. The same with removing the italicized good/correct. I think that the meaning is already clear to the reader.

Calling out the [IBM Style rule](https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#tense) to avoid future tense.

> Currently, the Operator does not reconcile when you update the `cloud-credentials` secret. This occurs because there are no {oadp-short} specific labels or owner references on the `cloud-credentials` secret. If you create a `cloud-credentials` secret with incorrect credentials, such as empty data, the Operator reconciles and creates a Backup Storage Location (BSL) and registry deployment with the empty data. As a result, when you update  `cloud-credentials` secret with the correct credentials, the Operator does not immediately reconcile to catch the new credentials. This issue is fixed in {oadp-short} 1.3.

Author: ```suggestion
Currently, the {oadp-short} Operator does not reconcile when you update the `cloud-credentials` secret. This occurs because there are no {oadp-short} specific labels or owner references on the `cloud-credentials` secret. If you create a `cloud-credentials` secret with incorrect credentials, such as empty data, the Operator reconciles and creates a Backup Storage Location (BSL) and registry deployment with the empty data. As a result, when you update  `cloud-credentials` secret with the correct credentials, the Operator does not immediately reconcile to catch the new credentials. As this issue is fixed in {oadp-short} 1.3,  the workaround is to update to {oadp-short}.
```

> Currently, the {oadp-short} Operator does not reconcile when you update the `cloud-credentials` secret. This occurs because there are no {oadp-short} specific labels or owner references on the `cloud-credentials` secret. If you create a `cloud-credentials` secret with incorrect credentials, such as empty data, the Operator reconciles and creates a Backup Storage Location (BSL) and registry deployment with the empty data. As a result, when you update  `cloud-credentials` secret with the correct credentials, the Operator does not immediately reconcile to catch the new credentials. As this issue is fixed in {oadp-short} 1.3,  the workaround is to update to {oadp-short} 1.3.

Reviewer: I noticed you have a couple of extra spaces, @anarnold97. Also, the SSG has a very simple format for workarounds. :)

If you are online and want to fix this really quick, I'll pick up the merge review.

```suggestion
Currently, the {oadp-short} Operator does not reconcile when you update the `cloud-credentials` secret. This occurs because there are no {oadp-short} specific labels or owner references on the `cloud-credentials` secret. If you create a `cloud-credentials` secret with incorrect credentials, such as empty data, the Operator reconciles and creates a Backup Storage Location (BSL) and registry deployment with the empty data. As a result, when you update the `cloud-credentials` secret with the correct credentials, the Operator does not immediately reconcile to catch the new credentials. 

Workaround: Update to {oadp-short} 1.3.
```

Reviewer: Sorry I missed the modifier _the_ on the first round. I was training a peer reviewer for the squad and got distracted.

[id="known-issues-1-2-3_{context}"]
== Known issues

The {oadp-short} 1.2.3 has the following known issue:

> . Data Protection Application (DPA) does not reconcile when the credentials secret is updated

Reviewer: When you fixed the abbreviation, an extra space was added, and now this renders as a numbered bullet.

![image](https://github.com/openshift/openshift-docs/assets/82119979/d9820040-ce8d-4ad4-8363-ab4308b1e84f)

As long as no other rendering errors creep in, this LGTM and I can merge.

> . Data Protection Application (DPA) does not reconcile when the credentials secret is updated

Author: ```suggestion
.Data Protection Application (DPA) does not reconcile when the credentials secret is updated
```
